app/views/contexts/summary/list_collector_block.py

```diff
@@ -0,0 +1,235 @@
+            item_label,
+            item_anchor,
+        ) = (None, None, None, None, None, None)
+        current_list = self._list_store[summary["for_list"]]
+
+        list_collector_blocks = list(
+            self._schema.get_list_collectors_for_list(
+                self._section, for_list=summary["for_list"]
+            )
+        )
+
+        add_link = self._add_link(summary, list_collector_block)
+
+        list_collector_blocks_on_path = [
+            list_collector_block
+            for list_collector_block in list_collector_blocks
+            if list_collector_block["id"] in self._routing_path.block_ids
+        ]
+
+        list_collector_block = (
+            list_collector_blocks_on_path[0]
+            if list_collector_blocks_on_path
+            else list_collector_blocks[0]
+        )
+
+        rendered_summary = self._placeholder_renderer.render(
+            summary, self._location.list_item_id
+        )
+
+        section_id = self._section["id"]
+        if list_collector_blocks_on_path:
+            edit_block_id = list_collector_block["edit_block"]["id"]
+            remove_block_id = list_collector_block["remove_block"]["id"]
+            add_link = self._add_link(summary, list_collector_block)
+            related_answers = self._get_related_answers(current_list)
+            item_anchor = self._schema.get_item_anchor_answer_id(
+                section_id, current_list.name
+            )
+            item_label = self._schema.get_item_label(section_id, current_list.name)
+
+        if len(current_list) == 1 and current_list.primary_person:
+            if primary_person_block := self._schema.get_list_collector_for_list(
+                self._section, for_list=summary["for_list"], primary=True
+            ):
+                primary_person_edit_block_id = edit_block_id = primary_person_block[
+                    "add_or_edit_block"
+                ]["id"]
+        list_summary_context = self.list_context(
+            list_collector_block["summary"],
+            for_list=list_collector_block["for_list"],
+            return_to="section-summary",
+            edit_block_id=edit_block_id,
+            remove_block_id=remove_block_id,
+            primary_person_edit_block_id=primary_person_edit_block_id,
+        )
+
+        return {
+            "title": rendered_summary["title"],
+            "type": rendered_summary["type"],
+            "add_link": add_link,
+            "add_link_text": rendered_summary["add_link_text"],
+            "empty_list_text": rendered_summary.get("empty_list_text"),
+            "list_name": rendered_summary["for_list"],
+            "related_answers": related_answers,
+            "item_label": item_label,
+            "item_anchor": item_anchor,
+            **list_summary_context,
+        }
+
+    @property
+    def list_context(self):  # type: ignore
+        # Ignore return type due to circular import warning in tests
+        return contexts.ListContext(
```

Comment:
This circular import can be avoided by importing directly from the files, instead of importing via `___init__.py`. Will need to change a couple of imports but it can work without the circular import. It is circular import due to the way we import the classes into __init__.py then reuse. The init style import should be only for external classes that import them, not internal class. Internal classes should import with absolute paths.